File: Archive of old scripts and tests/scripts_archive/ListFilesFun.R
getfiles <- function(folder, extension, get_prefix = TRUE) {
  # List all files recursively in the folder
  
  folder = wd
  extension = ".pep_calibrated.tsv"
  
  # List all files recursively in the folder with the specified extension
  all_files <- list.files(wd, pattern = extension, full.names = TRUE, recursive = TRUE)
  relative_paths <- 
  
    # Get relative paths
  relative_paths <- sapply(all_files, function(file) {
    normalizePath(file, winslash = "/", mustWork = FALSE)
  })
  
  # Initialize prefixes vector
  prefixes <- character()
  
  if (get_prefix) {
    # Extract prefixes
    prefixes <- sapply(relative_paths, function(path) {
      # Replace backslashes with underscores and remove the extension
      prefix <- gsub("\\\\", "_", path)
      prefix <- sub("_interact-ipro-ptm\\.pep\\.xml$", "", prefix)
      prefix <- sub("_interact-prob$", "", prefix)
      dirname(prefix)
    })
  }
  
  if (get_prefix) {
    return(list(relative_paths, prefixes))
  } else {
    return(relative_paths)
  }
}

# Example usage
# files_and_prefixes <- getfiles("path/to/folder", "xml")
# files_only <- getfiles("path/to/folder", "xml", FALSE)

File: 01_Sulfotyrosine_Data_Processing/scripts/ListFilesFun.py
import os

def getfiles(folder, extension, get_prefix=True, removal_strings=None):
    relative_paths = []
    prefixes = []
    # these are the filenames, so far i have come across these 3
    if removal_strings is None:
        removal_strings = ["_interact-ipro-ptm.pep.xml", "_interact-prob.pep.xml", "_interact-ipro.pep.xml"]

    for root, dirs, files in os.walk(folder):
        for file in files:
            if file.endswith(extension):
                relative_path = os.path.relpath(os.path.join(root, file), folder)
                relative_paths.append(relative_path)
                if get_prefix:
                    prefix = relative_path.replace("\\", "_")

                    # Remove specified strings if found in the prefix
                    for removal_string in removal_strings:
                        prefix = prefix.replace(removal_string, "")

                    # Strip the file extension if present
                    prefix, _ = os.path.splitext(prefix)

                    prefixes.append(prefix)

    return (relative_paths, prefixes) if get_prefix else relative_paths
